src/dbnode/client/session.go

```diff
@@ -255,7 +257,25 @@ func newSession(opts Options) (clientSession, error) {
 		},
 		metrics: newSessionMetrics(scope),
 	}
-	s.reattemptStreamBlocksFromPeersFn = s.streamBlocksReattemptFromPeers
+	s.reattemptStreamBlocksFromPeersFn = func(
+		reattemptBlocks []receivedBlockMetadata,
+		reEnqueueCh enqueueChannel,
+		err error,
+		reattemptReason reason,
+		reattemptType reattemptType,
+		m *streamFromPeersMetrics,
+	) {
+		if err := s.streamBlocksReattemptFromPeers(
```

Comment:
Capturing what we discussed, let's just make all errors propagate and perform the followup of retrying top level bootstrap.